Source/WebGPU/WebGPU/BindGroup.mm

```diff
@@ -924,28 +925,28 @@ static BindGroupEntryUsageData makeBindGroupEntryUsageData(BindGroupEntryUsage u
     constexpr ShaderStage stagesPlusUndefined[] = { ShaderStage::Vertex, ShaderStage::Fragment, ShaderStage::Compute, ShaderStage::Undefined };
     constexpr size_t stageCount = std::size(stages);
     constexpr size_t stagesPlusUndefinedCount = std::size(stagesPlusUndefined);
-    const auto& bindGroupLayout = WebGPU::fromAPI(descriptor.layout);
-    if (!bindGroupLayout.isValid() || (!bindGroupLayout.isAutoGenerated() && descriptor.entryCount != bindGroupLayout.entries().size()) || &bindGroupLayout.device() != this) {
+    Ref bindGroupLayout = WebGPU::protectedFromAPI(descriptor.layout);
```

Comment:
I just made the changes in this PR but is there / will there be a style checker rule to enforce that convention? It seems easy to overlook and would otherwise pass the static analysis warning.

Reply:
That's probably a good idea. Maybe we should add this to https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines as well. cc @cdumez